### ISSUE TITLE
Use Argon2i parameters from the passed PasswordHash in pwhash::hash_password_verify

### ISSUE THREAD

**Issue by vlmutolo** [brycx/orion#138]
I was using orion's `pwhash` module for my recipe website authentication process, and hit a small snag in the offered API. (Emphasis on "small"; I love the orion API.)

## TL;DR
I think `pwhash::hash_password_verify` should pull the Argon2i parameters from the `PasswordHash` passed to the function and not have the user pass them as arguments. Thus, the function signature would become:

```rust
hash_password_verify(password: &Password, hash: &PasswordHash) -> Result<(), UnknownCryptoError>
```

## Use Case
My use case is that I want to store a bunch of users' passwords in a database (using, of course, the very nice `unprotected_as_encoded function`). Then, when they log in, I want to pull the hash from the DB and compare it to the password they just provided. This has to be the most textbook example of using the `pwhash` module I can imagine.

However, I was thinking about what happens if I change the argon2i parameters for the hash. All the old password hashes would become invalidated. If the change isn't that necessary for security, a really good UX for the website user would be to just update the hash the next time they log in and provide a plaintext password. But this would require having password hashes in the database that were hashed with different Argon2i parameters, which would require me keeping the parameters in the database.

I could do that, but I also think this is a good opportunity to improve the orion API. The parameters are already stored in the encoded PasswordHash string, and in the PasswordHash itself. It would be trivial to just pull the parameters at the time of comparison, and pretty brain-dead for the developer using orion as a dependency.

### Before the change
```rust
use orion::pwhash;

let password = pwhash::Password::from_slice(b"Secret password")?;
let hash = pwhash::hash_password(&password, 3, 1<<16)?;

// Correct paramters
assert!(pwhash::hash_password_verify(&hash, &password, 3, 1<<16).is_ok());

// Incorrect parameters
assert!(pwhash::hash_password_verify(&hash, &password, 3, 1<<15).is_err());
assert!(pwhash::hash_password_verify(&hash, &password, 4, 1<<16).is_err());
```

### After the change
```rust
use orion::pwhash;

let password1 = pwhash::Password::from_slice(b"Secret password")?;
let password2 = pwhash::Password::from_slice(b"Secret password")?;

// Pretend these are stored somewhere and out-of-mind, e.g. in a database.
let hash1 = pwhash::hash_password(&password, 3, 1<<16)?;
let hash2 = pwhash::hash_password(&password, 4, 2<<16)?;

// We don't have to remember which password used what parameters when it
// time to verify them.
assert!(pwhash::hash_password_verify(&hash1, &password1).is_ok());
assert!(pwhash::hash_password_verify(&hash2, &password2).is_ok());
```

**Comment by brycx** [comment]
Once again, thank you for the very detailed and good use-case explanation! It's great to have it put so clearly.

As we discussed in the Matrix channel, I do believe this to be a very nice improvement to the `pwhash` API, that you have suggested. The example of the API change you present, makes this even clearer.

**Comment by brycx** [comment]
Fixed in #137.
